GitUI/UserControls/FileStatusList.cs

```diff
@@ -620,11 +620,34 @@ public void SetDiffs(IReadOnlyList<GitRevision> revisions)
                 GitRevision[] parentRevs;
                 if (revisions.Count == 1)
                 {
-                    // Note: RevisionGrid could in some forms be used to get the parent guids
+                    // Get the parents for the selected revision
                     parentRevs = Revision.ParentIds?.Select(item => new GitRevision(item)).ToArray();
                 }
+                else if (revisions.Count == 2 || revisions.Count > 4)
+                {
+                    // Assume only first to selected is interesting
+                    parentRevs = new GitRevision[1] { revisions.Last() };
+                    if (AppSettings.ShowDiffForAllParents)
+                    {
+                        // Get base commit
+                        Lazy<ObjectId> head = new Lazy<ObjectId>(() => Module.RevParse("HEAD"));
+                        var anyArtificial = parentRevs[0].IsArtificial || Revision.IsArtificial;
+                        var revA = parentRevs[0].IsArtificial ? head.Value : parentRevs[0].ObjectId;
+                        var revB = Revision.IsArtificial ? head.Value : Revision.ObjectId;
+                        ObjectId baseRevGuid = Module.GetMergeBase(revA, revB);
+                        if (baseRevGuid != null
+                            && baseRevGuid != revA
+                            && baseRevGuid != revB
+                            && (!anyArtificial || baseRevGuid != head.Value))
```

Comment:
I think, the last condition can be removed because it is redundant.

Reply:
This was to handle selection of work tree/index and did not work correctly (should show HEAD as BASE). Rewrote.